cli/src/commands/extensions/state.rs

```diff
@@ -0,0 +1,94 @@
+//! Shared extension state.
```

Comment:
A separate module here allows us to control access through visibility modifiers.